packages/jest-cli/src/TestRunner.js

```diff
@@ -304,6 +242,92 @@ class TestRunner {
       });
   }
 
+  _setupReporters(): Array<JestReporterBase> {
```

Comment:
this is where all reporters get set up.
after that each event will jest call the corresponding method on each of them